pytorch_lightning/trainer/training_loop.py

```diff
@@ -307,13 +307,16 @@ def has_arg(self, *args):
 
     def train(self):
         # add signal handlers for process kills
-        # def _signal_kill_handler(*args):
-        #     return TrainerTrainLoopMixin.run_training_teardown(self)
-        #
-        # orig_signal_handlers = {}
-        # for sig_name in SIGNAL_TERMINATE:
-        #     orig_signal_handlers[sig_name] = signal.signal(getattr(signal, sig_name),
-        #                                                    _signal_kill_handler)
+        def _signal_kill_handler(sig, name):
+            return self.run_training_teardown()
+
+        atexit.register(self.run_training_teardown)
+
+        orig_signal_handlers = {}
+        for sig_name in SIGNAL_TERMINATE:
```

Comment:
Can we rename SIGNAL_TERMINATE to FATAL_SIGNALS or similar. Originally I thought this was just sigterm.